### DISCUSSION TITLE
Advice on UnnecessaryAbstractClass when used to prevent instantiation

### DISCUSSION THREAD

**Discussion by TWiStErRob** [detekt/detekt#5382] [Q&A] [answered]
Consider this hierarchy:
```mermaid
classDiagram
    BaseNode <|-- Historical
    BaseNode <|-- User
    Historical <|-- Film
    Historical <|-- Cinema
    BaseNode: +Long graphId
    User: +String email
    Historical: OffsetDateTime created
    Historical: OffsetDateTime modified
    Historical: OffsetDateTime deleted
    class Cinema {
        +String name
    }
    class Film {
        +String name
        +LocalDate released
    }
```

In this case I would make `BaseNode` and `Historical` `abstract`, because they're not meant to be used directly. No-one should be able to instantiate `Historical()` because it doesn't mean anything, but since there are multiple `Historical` entities, the base class makes sense for polymorphism.

Now, Detekt flags this:
```
UnnecessaryAbstractClass - [An abstract class without an abstract member can be refactored to a concrete cla(...)] at Historical.kt:14:16
```
What is the recommended course of action in this case?

**Comment by BraisGabin** [comment, marked as answer]
I would vote for a sealed class or an interface in this case. But this could be a good case for a `Suppress`. That class could be a concrete one but you prefer it to be abstract.
